src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.nikita.chiken_bell.core.service.impl;
+
+import org.nikita.chiken_bell.core.entity.Customer;
+import org.nikita.chiken_bell.core.exception.NotFoundCustomerException;
+import org.nikita.chiken_bell.core.exception.PhoneExistException;
+import org.nikita.chiken_bell.core.service.CustomerService;
+
+import java.util.*;
+
+public class CustomerServiceImpl implements CustomerService {
+
+    private final Set<Customer> customers = new HashSet<>();
+
+
+    @Override
+    public Customer create(String name, String phone, String address) {
+        if (customers.stream().filter(x -> x.getPhone().equals(phone)).findFirst().isPresent()){
+            throw new PhoneExistException();
+        }
+        Customer customer = new Customer(name, phone, address);
+        customers.add(customer);
+        return customer;
+    }
+
+    @Override
+    public Customer update(String id, String name, String phone, String address) {
+        Customer customer =getById(id).orElseThrow(NotFoundCustomerException::new);
+        customer.setName(name);
+        customer.setPhone(phone);
```

Comment:
1. we need check, that phone is not null
2.  we need check, that phone is unique

Reply:
What we need do , if phone null ?